include/boost/url/impl/url_view.ipp

```diff
@@ -456,46 +456,50 @@ url_view::
 compare(const url_view& other) const noexcept
 {
     int comp = detail::ci_compare(
-        scheme(),
-        other.scheme());
+        get(id_scheme),
```

Comment:
I missed last time we need the complete thing (`get(...)`) to ensure we eliminate some false positives.

Reply:
did you add a test? Is this for hash or is this for compare?

Reply:
It's for compare. The previous tests are still valid. It's just a matter of whether the surrounding reserved chars are considered in the comparison. There weren't considered but I realized they need to be considered.